collie/model/hybrid_matrix_factorization.py

```diff
@@ -207,38 +239,102 @@ def __init__(self,
+                    {
+                        'lr': metadata_only_stage_lr,
+                        'optimizer': metadata_only_stage_optimizer,
+                        # optimize metadata layers only
+                        'parameter_prefix_list': [
+                            'user_metadata', 'combined', 'user_bias', 'item_bias'
+                        ],
+                        'stage': 'user_metadata_only',
+                    },
+                    {
+                        'lr': all_stage_lr,
+                        'optimizer': all_stage_optimizer,
+                        # optimize everything
+                        'parameter_prefix_list': ['user', 'item', 'user_metadata', 'combined'],
+                        'stage': 'all',
+                    },
+                ]
+            elif item_metadata is not None:
+                optimizer_config_list = initial_optimizer_block + [
+                    {
+                        'lr': metadata_only_stage_lr,
+                        'optimizer': metadata_only_stage_optimizer,
+                        # optimize metadata layers only
+                        'parameter_prefix_list': [
+                            'item_metadata', 'combined', 'user_bias', 'item_bias'
+                        ],
+                        'stage': 'item_metadata_only',
+                    },
+                    {
+                        'lr': all_stage_lr,
+                        'optimizer': all_stage_optimizer,
+                        # optimize everything
+                        'parameter_prefix_list': ['user', 'item', 'item_metadata', 'combined'],
+                        'stage': 'all',
+                    },
+                ]
         super().__init__(optimizer_config_list=optimizer_config_list,
                          item_metadata_num_cols=item_metadata_num_cols,
+                         user_metadata_num_cols=user_metadata_num_cols,
                          **get_init_arguments())
 
     __doc__ = merge_docstrings(MultiStagePipeline, __doc__, __init__)
 
     def _move_any_external_data_to_device(self):
-        """Move item metadata to the device before training."""
-        self.item_metadata = self.item_metadata.to(self.device)
+        """Move item and user metadata to the device before training."""
+        if self.item_metadata is not None:
+            self.item_metadata = self.item_metadata.to(self.device)
+        if self.user_metadata is not None:
+            self.user_metadata = self.user_metadata.to(self.device)
 
     def _load_model_init_helper(self, load_model_path: str, map_location: str, **kwargs) -> None:
-        self.item_metadata = (
-            joblib.load(os.path.join(load_model_path, 'metadata.pkl'))
-        )
+        try:
+            self.item_metadata = (
+                joblib.load(os.path.join(load_model_path, 'item_metadata.pkl'))
+            )
+        except FileNotFoundError:
+            warnings.warn('`item_metadata.pkl` not found')
```

Comment:
[Optional] I have been burned by single-ticks in the past and try to use double-ticks everywhere in the repo where I can. 

```suggestion
            warnings.warn('``tem_metadata.pkl`` not found')
```

Reply:
👍